app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1125,6 +1120,7 @@ private void attemptDispatchHomeScreen() {
             handlePendingSync();
         } else if (CommCareApplication._().isConsumerApp() && !sessionNavigationProceedingAfterOnResume) {
             // so that the user never sees the real home screen in a consumer app
+            sessionNavigationProceedingAfterOnResume = false;
```

Comment:
the fact that I was missing this statement is what was causing the weird activity navigation bugs, although to be honest I am not sure how it manifested the way it did.